[31기 윤재웅]ADD:기능구현/리팩토링 완료 (추가기능 7번 제외)

JaySurplusYoon opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [ ] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표
- 아이디, 비밀번호 입력창을 사용한 로그인 기능 구현
- 포스팅 하단에 댓글입력/삭제/좋아요 기능 구현
- 프로필 클릭시 메뉴박스 노출 가능 구현
- 화면 리사이징에 따른 반응형 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
1. [Mission 1]로그인 데이터 입력여부 검사
- 입력창에 변화가 일어날 때 마다 유효성 검사 함수가 실행.
- 유효성 검사 함수의 반환값(boolean)에 따라 버튼의 배경색 변환.
- 테스트 방법 : `/login` 페이지 이동 >>> id/pwd input에 최소 1글자 이상 데이터 입력

2. [Mission 5] 로그인 제출값 유효성 검사 
- 로그인 입력후 enter 혹은 버튼 클릭시마다 유효성 검사 함수 실행.
- 유효할 경우에는 main.html로 이동, 그 외에는 alert로 유효한 값 입력하도록 안내.
- 테스트방법 : `/login` 페이지 이동 >>> id/pwd input에 아이디에 '@'포함, 패스워드는 5글자 이상 입력시에만 로그인 허용

3.[Mission 4, Mission 6]댓글 내용 추가/제거/좋아요 기능
- 댓글입력 후 Enter 혹은 입력버튼 클릭시 댓글 입력 함수 실행.
- 댓글창 컨테이너의 클릭을 감지하여 클릭한 지점의 클래스명을 파악하고 댓글삭제/좋아요 함수 각각 실행.
- 테스트방법: /main페이지 이동>>>피드사진 하단에 댓글 입력>>>입력한 댓글 우측의 X,♥️ 아이콘 클릭하여 반응여부 확인.

4.[Mission 8]nav 프로필 사진 클릭 시 메뉴 박스 생성
- nav상단에 프로필사진 클릭시 화면 최상단에 메뉴박스 노출.
- 메뉴박스에 마우스오버시 배경색 변환
- 메뉴박스가 나타난 상태에서 화면전체의 클릭감지시 메뉴박스 invisible처리하여 제거.
- 테스트방법: /main페이지 이동>>>nav바 상단 우측의 프로필 사진 클릭 >>>박스의 각 메뉴에 마우스오버 >>> 화면 어디든 클릭해서 박스 제거여부 확인.

5.[Mission 9]반응형 구현
- 화면 리사이즈시 상단nav바의 아이콘 위치를 준용하여, 피드가 담긴 section영역의 왼쪽의 간격 조정.
- 가로폭 일정크기 미만시 우측의 개인추천영역 제거하는 기능 구현
- 테스트방법:/main페이지 이동 >>>화면 width를 인위적으로 조정하여,피드화면 이동여부 및 개인추천영역 제거여부 확인

## :: 성장 포인트
1)코드최초작성과정
- 레이아웃과 js기능을 같이 만들며 가다보니 당초 예상한 일정보다 완성이 늦어진다는 점 확인. 향후에는 레이아웃부터 진행하도록 할 것.
- flex와 position 기능숙지
- 화살표 함수 작성방법 숙지
- eventListener활용방법 숙지

2)리팩토링과정

- 코드 전체 리뷰하며 과도한 div생성 및 class지정한 부분 제거.
- bottom-up과정에서 불필요한 div와 정렬요소, width , height 제거.
- 모든 클래스명을 독립적으로 작성해서 리팩토림 어려웠음. 추후에는 같은 속성의 화면요소를 모아 범용적인 클래스명을 지정한 후 html작성하고자 함  (as is : my-thumbnail, follower-thumbnail  => to be: common-thumbnail)
- 함수를 기능단위로 잘게 쪼갬으로써 유지보수 용이하도록 하겠음.

<br />

## :: 기타 질문 및 특이 사항
- [MISSION05]로그인 유효성검사시  input type을 각각 email과 password로 변경하여 js없이 html자체로만 제어하려고 하였으나 실패하였습니다. 잘못된 부분을 알려주시면 감사하겠습니다.
- 가이드 주신 bottom-up방식으로 css를 리팩토링하긴 했으나, 가독성이 여전히 떨어집니다. 제가 작성한 방식이 맞는지 확인 요청드립니다.
- master에서 PR을 받아 검토한 후 merge하는 일련의 과정이 궁금합니다.

yojuyoon commented on students/18th/kimhyunjoong/js/main.js:
```diff
@@ -0,0 +1,148 @@
+  commentLeft.appendChild(commentContent);
+  commentRight.appendChild(deleteBtn);
+  commentRight.appendChild(heartBtn);
+  comment.appendChild(commentLeft);
+  comment.appendChild(commentRight);
```
위 리뷰 내용 확인해주세요!

hyjoong replied:
네 알겠습니다 !!!!!

yojuyoon commented on students/18th/kimhyunjoong/main.html:
```diff
@@ -0,0 +1,275 @@
+      <div class="mainWrapper">
+        <section class="feeds">
+          <div class="storyBox">
+            <div class="story">
+              <div class="storyPicture">
+                <img
+                  src="https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-19/s150x150/19120841_176332809566752_3268745209206276096_n.jpg?_nc_ht=scontent-ssn1-1.cdninstagram.com&amp;_nc_ohc=RbJaaEJjcgwAX-5SRSs&amp;tp=1&amp;oh=0b77b03bc1d68bacbefa3f582cad2901&amp;oe=605EA7BD"
+                />
+              </div>
+              <div class="story-nikname">Hyunjoong</div>
+            </div>
+
+            <div class="story">
+              <div class="storyPicture">
+                <img
+                  src="https://scontent-ssn1-1.cdninstagram.com/v/t51.2885-19/s150x150/131239636_863191087766011_5065572424909332970_n.jpg?_nc_ht=scontent-ssn1-1.cdninstagram.com&amp;_nc_ohc=DcIgeapDfV0AX_6zfWJ&amp;tp=1&amp;oh=cb1c05e76ada1ec27bc6ce8470894f61&amp;oe=605E66C5"
+                />
+              </div>
+              <div class="story-nikname">ham_s1</div>
+            </div>
+
+            <div class="story">
+              <div class="storyPicture">
+                <img
+                  src="https://scontent.cdninstagram.com/v/t51.2885-19/44884218_345707102882519_2446069589734326272_n.jpg?_nc_ht=scontent.cdninstagram.com&amp;_nc_ohc=8AUoZNYMP5IAX81KVqq&amp;oh=5cb6c6a9bbf7568658d8c157b1607313&amp;oe=605FE68F&amp;ig_cache_key=YW5vbnltb3VzX3Byb2ZpbGVfcGlj.2"
+                />
+              </div>
+              <div class="story-nikname">khb_2211</div>
+            </div>
```
추후 반복되는 구조는 데이터를 따로 분리한 뒤 배열 메서드를 활용해서 구조를 축약해주는 방법이 있습니다!

hyjoong replied:
오호 따로 공부 해보겠습니다 !

yojuyoon reviewed (commented):
현중님! 기능 구현 잘 해주셨네요! 고생많으셨습니다! 리뷰 내용 확인하셔서 리팩토링 할 때 참고해주세요! 🙌

Tsple commented on students/18th/kimhyunjoong/js/login.js:
```diff
@@ -0,0 +1,45 @@
+  }
+  if (!loginPassword) {
+    unActiveBtn();
+  }
```
수정 했습니다

Tsple commented on students/18th/kimhyunjoong/js/login.js:
```diff
@@ -0,0 +1,45 @@
+function unActiveBtn() {
+  loginBtn.style.z = skyBlue;
+  loginBtn.disabled = true;
+}
```
수정 했습니다

Tsple commented on students/18th/kimhyunjoong/js/login.js:
```diff
@@ -0,0 +1,39 @@
+});
+
+loginPassword.addEventListener("keyup", () => {
+  const loginEmail = document.getElementById("loginEmail").value;
+  const loginPassword = document.getElementById("loginPassword").value;
+  loginEmail.includes("@") && loginPassword.length >= 5
+    ? onActiveBtn()
+    : unActiveBtn();
+});
```
수정 했습니다!

Tsple commented on students/18th/kimhyunjoong/js/login.js:
```diff
@@ -0,0 +1,39 @@
+});
+
+function onActiveBtn() {
+  loginBtn.style.backgroundColor = deepBlue;
+  loginBtn.disabled = false;
+}
+
+function unActiveBtn() {
+  loginBtn.style.backgroundColor = skyBlue;
+  loginBtn.disabled = true;
+}
```
수정 했습니다~

Tsple commented on students/18th/kimhyunjoong/main.html:
```diff
@@ -0,0 +1,275 @@
+    <title>Westagram</title>
+  </head>
+  <body>
+    <nav class="nav">
```
굿 ~

hyjoong commented on students/18th/kimhyunjoong/main.html:
```diff
@@ -0,0 +1,275 @@
+  </head>
+  <body>
+    <nav class="nav">
+      <div class="navLeft">
+        <a href="main.html"> Instagram </a>
+      </div>
+      <div class="navCenter">
+        <input type="text" placeholder="검색" />
+        <i class="fas fa-search"></i>
+      </div>
+      <div class="navRight">
```
네 알겠습니다 !

JaySurplusYoon committed:
[31기 윤재웅]프로젝트 개시

JaySurplusYoon committed:
ADD:로그인 기능 고도화/ RENAME:index.html->login.js

JaySurplusYoon committed:
feat:댓글입력기능 추가

JaySurplusYoon committed:
feat:댓글삭제 및 댓글좋아요/안좋아요 기능 추가

JaySurplusYoon committed:
feat : 프로필아이콘 클릭시 나타나는 프로필창 구현
 - 마우스오버에 따른 버튼색상 변경
 - 프로필아이콘 외의 영역 클릭시 프로필창 제거거

JaySurplusYoon committed:
Fix:프로필아이콘 영역 클릭시 프로필창 사라지지않는 오류 수정

JaySurplusYoon committed:
fix:리팩토링
 - console.log제거
 - 내어쓰기 2줄로 변경
 - css 클래스명 변경(asis:케밥 tobe:카멜)
 - bottomup방식 css 재배치 및 불필요한 클래스 제거

JaySurplusYoon committed:
fix:로그인 유효성검사 기능 수정
